src/actions/contractActions.ts

```diff
@@ -160,19 +161,14 @@ export function fetchContract(hash: string, populateStates = true) {
   ): Promise<void> => {
     if (shouldFetchContract(getState(), hash)) {
       dispatch(requestContract(hash))
-
+      const [network, protocol] = getNetworkAndProtocol()
       try {
-        const response = await fetch(`${GENERATE_BASE_URL()}/contract/${hash}`)
-
-        const json = await response.json()
+        const restAPI = protocol === 'neo3' ? NeoRest : NeoLegacyREST
+        const json: any = await restAPI.contract(hash, network)
```

Comment:
the `any` type was necessary because otherwise it would complain at this piece a couple lines below
```
json.invocationStats = invocationStats || null
```
because the response doesn't have a `invocationStats` field. 
I guess ideally `requestContractSuccess()` should consume the types from `dora-ts` instead of it's own, but yeah that can be a future enhancement.

Reply:
Can we just use the optional type here for that field => https://www.typescriptlang.org/docs/handbook/interfaces.html#optional-properties ??

Also I totally agree that types should be exported from the sdk

Reply:
The field is originally not on the response object from the `restAPI.contract()` call (being [this for N3](https://github.com/CityOfZion/dora-sdk-ts/blob/dc462350340fcd60c06c0f96064bbb7007a1e7a9/src/interfaces/api/neo/responses.ts#L74) and [this for N2](https://github.com/CityOfZion/dora-sdk-ts/blob/dc462350340fcd60c06c0f96064bbb7007a1e7a9/src/interfaces/api/neo_legacy/interface.ts#L33)). It is just data from another RPC call glued into the previous response object because it probably needs this at some point. Even the resulting data in the `json` variable that is passed to this 

https://github.com/CityOfZion/dora/blob/778cdc51b4b7ce8158fade6fd25a8125e6d450ef/src/actions/contractActions.ts#L35-L36 

doesn't even match the `Contract` type. The most obvious one is the `invocationStats` property that doesn't exist on it. It seems to somehow work but I fail to understand how this can be type safe 😅 

tldr; i can't make `invocationStats` optional because it didn't exist on the response object to being with.